C7FIT/MapViewController.swift

```diff
@@ -0,0 +1,172 @@
+        
+        seconds = 0.0
+        distance = 0.0
+        locations.removeAll(keepingCapacity: false)
+        timer = Timer.scheduledTimer(timeInterval: 1, target: self, selector: #selector(eachSecond), userInfo: nil, repeats: true)
+        setupLocationTracking()
+    }
+    
+    func stopTrackRun(){
+        mapViewApp.startButton.isEnabled = true
+        mapViewApp.stopButton.isEnabled = false
+        
+        //open up run in detail view here
+        //ask user if they want to save run
+        createRunData()
+        self.navigationController?.pushViewController(MapDetailViewController(run:lastRun), animated: true)
+        timer.invalidate()
+        locationManager.stopUpdatingLocation()
+    }
+    
+    // MARK: Create Run Data
+    func createRunData() {
+        var tempRun = RunData()
+        var savedLocations = [Location]()
+        
+        print(savedLocations)
+        var i = 0
+        for location in locations {
+            var tempLocation = Location()
+            tempLocation.timestamp = location.timestamp.description
+            tempLocation.latitude = location.coordinate.latitude
+            tempLocation.longitude = location.coordinate.longitude
+            savedLocations.append(tempLocation)
+            i = i + 1
+        }
+        tempRun.distance = distance
+        tempRun.time = seconds
+        tempRun.pace = pace
+        tempRun.locations = savedLocations
+        lastRun = tempRun
+    }
+    
+    // MARK: location tracking
+    func setupLocationTracking() {
+        locationManager.delegate = self
+        locationManager.desiredAccuracy = kCLLocationAccuracyNearestTenMeters
+        locationManager.activityType = .fitness
+        locationManager.distanceFilter = 5
+        locationManager.requestAlwaysAuthorization()
+        if (CLLocationManager.locationServicesEnabled()) {
+            locationManager.startUpdatingLocation()
+        }
+    }
+    
+    func locationManager(_ manager: CLLocationManager, didUpdateLocations locations: [CLLocation]) {
+        for location in locations {
+            //Focus on the runner
+            let center = CLLocationCoordinate2D(latitude: location.coordinate.latitude, longitude: location.coordinate.longitude)
+            let region = MKCoordinateRegion(center: center, span: MKCoordinateSpan(latitudeDelta: 0.01, longitudeDelta: 0.01))
+            mapViewApp.mapView.setRegion(region, animated: true)
+            
+            if location.horizontalAccuracy < 50 {
+                if self.locations.count > 0 {
+                    distance += location.distance(from: self.locations.last!)
+                }
+                self.locations.append(location)
+            }
+        }
+    }
+    
+    func locationManager(_ manager: CLLocationManager, didFailWithError error: Error)
+    {
+        print("Error \(error)")
+    }
+    
+    // MARK: Timer
+    func eachSecond(timer: Timer) {
+        seconds = seconds + 1
```

Comment:
I don't think this will be very accurate, Timers are not guaranteed to fire accurately -- so this will drift

in startTrackRun capture a Date for the current time and use its timeIntervalSinceReferenceDate property against a current time to determine seconds